test_util/test_upgrade_vpc.py

```diff
@@ -247,7 +247,7 @@ def setup_cluster_workload(self, dcos_api: DcosApiSession, healthcheck_app: dict
             # the master's view after the upgrade.
             # See this issue for why we check for a difference:
             # https://issues.apache.org/jira/browse/MESOS-1718
-            self.task_state_start = self.get_master_task_state(dcos_api, self.tasks_start[0])
+            self.task_state_start = self.get_master_task_state(dcos_api, self.tasks_start[self.test_app_ids[0]])
```

Comment:
This could use a comment explaining why we only check a single task. After reading this code and the linked ticket, it's not clear to me why we don't want to check all test app tasks to make sure the master's view of them didn't change.